### PR TITLE
Updates to the pull request template

### PR DESCRIPTION
**Pull request by mdbtucker**
# Description

Update the pull request template, as discussed with Joe.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -32,8 +34,6 @@ Links and Details can be found in the samples WG Teams Files.
 - [ ] Adhere to readme template 
 - [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
 - [ ] Enforce format via clang-format config file
-- [ ] Compile code using compiler flags and fix anything detected "enable /Wall -Wformat-security -Werror=format-security"
```
Comment (JoeOster) [inline .github/PULL_REQUEST_TEMPLATE.md:35]: is there a reason you are removing this?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -32,8 +34,6 @@ Links and Details can be found in the samples WG Teams Files.
...
-- [ ] Review Sample with Domain Expert: <insert NameHere>
```
Comment (JoeOster) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: is there a reason you are removing this?

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mdbtucker) [reply .github/PULL_REQUEST_TEMPLATE.md:35]: Nope, not sure what happened there, sorry that was sloppy of me.  Re-inserted them.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mdbtucker) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: Removed accidentally, re-inserted now.

**Review by JoeOster — APPROVED** [review]

**Pull request merged**